packages/blade/src/components/Input/TextInput/TextInput.tsx

```diff
@@ -340,10 +342,21 @@ const _TextInput: React.ForwardRefRenderFunction<BladeElementRef, TextInputProps
       );
     }
 
+    if (showClearButton && hasTrailingInteractionElement) {
+      return (
+        <BaseBox display="flex" gap="spacing.3">
```

Comment:
Let's actually add unit tests for these if not already present

Reply:
https://github.com/razorpay/blade/issues/2763